src/cap32.cpp

```diff
@@ -3330,7 +3335,25 @@ void set_osd_message(const std::string& message) {
    osd_message = " " + message;
 }
 
+#ifdef HAVE_PNG
+void dumpScreen(void) {
+   static int dump_num=0;
+   struct stat _stat;
 
+   if (stat(CPC.sdump_dir.c_str(), &_stat) == 0) {
+      if (S_ISDIR(_stat.st_mode)) {
+         SDL_Surface* shot;
+         shot = SDL_PNGFormatAlpha(back_surface);
+         std::string dumpPath = CPC.sdump_dir + "/" + std::string("dump") + std::to_string(dump_num) + ".png";
```

Comment:
You could avoid overwriting an existing snapshot (from previous session) and handle incrementing dump_num at the same time here by doing something like:

std::string dumpPath;
do
{
  dumpPath = ...;
  dump_num++;
} while (access(dumpPath.c_str(), F_OK) == 0)